SUSE/connect-ng
# POC for the new Collector interface

## Description
Related card : https://trello.com/c/u1pb12AM/3245-teddpoc-rfc-new-approach-for-telemetry-data-on-suseconnect-ng

This is a POC for the refactored new implementation for the metric collection interface. This is intended to improve on the current hw_infos metrics and make it more dynamic. 

Requirements can be found here: https://etherpad.opensuse.org/p/suseconnect-collector-framework

## How to test
* Check the code to see if the new implementation is easy to understand and it makes sense as an improvement
* Run the tests for the collectors package : `go test ./internal/collectors/`

Thanks for the review!

PS: Be aware this pull request is against master right now, to run the CI pipelines. It depends on: https://github.com/SUSE/connect-ng/pull/218 Please review this first!
